pathd/path_pcep.h

```diff
@@ -80,10 +81,41 @@
 		}                                                              \
 	} while (0)
 
+struct pcep_config_group_opts {
+	char name[64];
+	char tcp_md5_auth[TCP_MD5SIG_MAXKEYLEN];
+	bool draft07;
+	bool pce_initiated;
+	int keep_alive_seconds;
+	int min_keep_alive_seconds;
+	int max_keep_alive_seconds;
+	int dead_timer_seconds;
+	int min_dead_timer_seconds;
+	int max_dead_timer_seconds;
+	int pcep_request_time_seconds;
+	int state_timeout_inteval_seconds;
+};
 struct pce_opts {
 	struct ipaddr addr;
 	short port;
-	bool draft07;
+	char pce_name[64];
+	char config_group_name[64];
+	/* These are the values configured in the pce sub-commands,
+	 * these need to be stored for later merging. Notice, it could
+	 * be that not all of them are set. */
+	struct pcep_config_group_opts config_opts;
+	/* These are the values that should be used, and all values will be set.
+	 * They are a merge of the default values, optional config_group
+	 * values (which overwrite default values), and any values configured
+	 * in the pce sub-commands (which overwrite both default and
+	 * config_group values). */
+	struct pcep_config_group_opts merged_opts;
```

Comment:
I don't feel like these processed fields belong in this data structure. This is supposed to represent the options given by the "user", not the state of option processing. Same thing for `is_best` and `previous_best`, it belong in some other data structure.

Reply:
The config_opts are indeed the values given by the user, and the merged_opts are just the combination of the default_values, the values referenced by the config_group_name, and the config_opts. This is the logical place to store these values. The is_best and previous_best were from Javier's changes for Multi-PCE, I'll remove them.

Reply:
By user I meant the user of the PCC API, and that is the vtysh code in this case. The fact that some options are given by configuration and some others are defaults or comes from other configuration group is an implementation detail of the vtysh code, I don't feel like it should be something the controller and pcc should have to know about and this struct is what is given to them.